pkg/apis/core/types.go

```diff
@@ -2171,6 +2177,85 @@ const (
 	VmPowerSpecSuspended VmPowerSpec = "suspended"
 )
 
+// Enables fine grained control of the block device mapping for an instance
+// This is typically used for booting servers from volumes
+type BlockDeviceMappingV2 struct {
+	BootIndex           int64
+	DeleteOnTermination bool
+	DestinationType     string
+	DeviceName          string
+	DeviceType          string
+	DiskBus             string
+	GuestFormat         string
+	NoDevice            bool
+	SourceType          string
+	Uuid                string
+	VolumeSize          int64
+	Tag                 string
+	VolumeType          string
+}
+
+type OpenStackPersonality struct {
+	Path     string
+	Contents string
+}
+
+type OpenStackSecurityGroup struct {
+	Name string
+}
+
+type GlobalScheduling struct {
+	// IPv4 address that should be used to access this server
+	// +optional
+	AccessIpv4 string
+	// IPv6 address that should be used to access this serve
+	// +ooptional
+	AccessIpv6 string
+	// The administrative password of the server
```

Comment:
Is this a plain password? If yes, we can't persist it without encryption.

Reply:
![image](https://user-images.githubusercontent.com/19650434/88719990-02b33100-d0d9-11ea-9500-9a73560c0958.png)
This is the definition from OpenStack Doc.

The image below is the example:
![image](https://user-images.githubusercontent.com/19650434/88720150-3ee69180-d0d9-11ea-8e25-95661c586428.png)

Reply:
Sorry, I thought you were mentioning the AccessIpv4 and AccessIpv6. AdminPass is used to log in to the OpenStack server without key pair. If so, we need to encrypt the value of this field and store it into ETCD, right?